hack/generator/pkg/codegen/pipeline/status_augment.go

```diff
@@ -63,32 +64,37 @@ func fuseAugmenters(augments ...augmenter) augmenter {
 
 // flattenAugmenter copies across the "flatten" property from Swagger
 func flattenAugmenter(allTypes astmodel.ReadonlyTypes) augmenter {
+	/* TODO: there is a lot here that should be pulled into a "default augmenter" value that can be reused,
+	but at the moment we only have one augmenter */
+
 	return func(main astmodel.Type, swagger astmodel.Type) (astmodel.Type, error) {
 		merger := astmodel.NewTypeMerger(func(ctx interface{}, left, right astmodel.Type) (astmodel.Type, error) {
 			// as a fallback, return left (= main) if we have nothing else to do
 			return left, nil
 		})
 
-		merger.Add(func(main, swagger astmodel.TypeName) (astmodel.Type, error) {
-			// when we merge two typenames we know that (structurally) they must
-			// be the ‘same’ type, even if they have different names
+		// need to resolve main type
+		merger.Add(func(main astmodel.TypeName, swagger astmodel.Type) (astmodel.Type, error) {
+			// Connascence alert! This is in coöperation with the code in
+			// determine_resource_ownership.go:updateChildResourceDefinitionsWithOwner
+			// which requires the typenames to identify which ChildResources match
+			// which Resources. The ChildResource types should never actually be
+			// used, so it's okay that we don't visit them here.
+			if strings.HasSuffix(main.Name(), "ChildResource") {
```

Comment:
minor: You could stick this in a const so that somebody interested could also find all usages of the cost which in theory would take them to the other location as well (although I think the comment is great)